pkg/app/piped/cloudprovider/lambda/function.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lambda
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	versionV1Beta1       = "pipecd.dev/v1beta1"
+	functionManifestKind = "LambdaFunction"
+)
+
+type FunctionManifest struct {
+	Kind       string               `json:"kind"`
+	APIVersion string               `json:"apiVersion,omitempty"`
+	Spec       FunctionManifestSpec `json:"spec"`
+}
+
+func (fm *FunctionManifest) validate() error {
+	if fm.APIVersion != versionV1Beta1 {
+		return fmt.Errorf("unsupported version: %s", fm.APIVersion)
+	}
+	if fm.Kind != functionManifestKind {
+		return fmt.Errorf("invalid manifest kind given: %s", fm.Kind)
+	}
+	if err := fm.Spec.validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// FunctionManifestSpec contains configuration for LambdaFunction.
+type FunctionManifestSpec struct {
+	Name     string            `json:"name"`
+	ImageURI string            `json:"image"`
```

Comment:
It may be more appropriate to replace it with  `ImageRef` that clearly represents a specific image that has a unique digest. But it's up to you.

Reply:
I think that name may confuse ( overlap with #1358 `ImageRef` struct ), otherwise this name specific exactly what I want to be given from here ( the URI to image on ecr registry ) so I think it's good for me 😅

Reply:
Okay, no objection to that 👍